src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    private static Map<Class<?>, MachineProducer> producerMap = new HashMap();
+
+    static {
+        producerMap.put(Bulldozer.class, new BulldozerProducer());
+        producerMap.put(Excavator.class, new ExcavatorProducer());
+        producerMap.put(Truck.class, new TruckProducer());
+    }
```

Comment:
this is okay, just FYI, in the real world this would likely be done with Map.of()